patches/0055-Modify-pthread_robust-source-to-work-with-uktest.patch

```diff
@@ -0,0 +1,80 @@
+From bc7ae7a6021ddd80ddfd3b91633d1772189b1f12 Mon Sep 17 00:00:00 2001
+From: Florin Postolache <florin.postolache80@gmail.com>
+Date: Sun, 23 Oct 2022 21:42:02 +0300
+Subject: [PATCH] Modify pthread_robust source to work with uktest
+
+Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>
+---
+ src/functional/pthread_robust.c | 25 ++++++++++++-------------
+ 1 file changed, 12 insertions(+), 13 deletions(-)
+
+diff --git a/src/functional/pthread_robust.c b/src/functional/pthread_robust.c
+index 44f846f..7444a33 100644
+--- a/src/functional/pthread_robust.c
++++ b/src/functional/pthread_robust.c
+@@ -1,12 +1,10 @@
+ #include <pthread.h>
+ #include <string.h>
+ #include <errno.h>
+-#include "test.h"
++#include <uk/test.h>
+ 
+-#define TEST(r, f, m) ( \
+-	((r) = (f)) == 0 || (t_error("%s failed: (pshared==%d, pi==%d) %s (" m ")\n", #f, pshared, pi, strerror(r)), 0) )
+-#define TESTX(r, f, x, m) ( \
+-	((r) = (f)) == (x) || (t_error("%s failed: (pshared==%d, pi==%d) got %d \"%s\" want %d \"%s\" (" m ")\n", #f, pshared, pi, r, strerror(r), x, strerror(x)), 0) )
++#define TEST(r, f, m) if ((f) != 0) return -1;
++#define TESTX(r, f, x, m) if ((f) != (x)) return -1;
+ 
+ static void *start_lock(void *arg)
+ {
+@@ -23,7 +21,7 @@ static void *start_wait(void *arg)
+ 	return 0;
+ }
+ 
+-void f(int pshared, int pi)
++int f(int pshared, int pi)
+ {
+ 	pthread_t td;
+ 	int r;
+@@ -31,7 +29,7 @@ void f(int pshared, int pi)
+ 	pthread_barrier_t barrier2;
+ 	pthread_mutexattr_t mtx_a;
+ 	pthread_mutex_t mtx;
+-
++	for(int pshared=0; pshared<=1; pshared++) for(int pi=0; pi<=1; pi++){
+ 	TEST(r, pthread_barrier_init(&barrier2, 0, 2), "creating barrier");
+ 
+ 	/* Robust mutexes */
+@@ -63,21 +61,22 @@ void f(int pshared, int pi)
+ 	TEST(r, pthread_mutex_init(&mtx, &mtx_a), "");
+ 	TEST(r, pthread_create(&td, 0, start_wait, (void *[]){ &barrier2, &mtx }), "");
+ 	r = pthread_barrier_wait(&barrier2);
+-	if (r && r != PTHREAD_BARRIER_SERIAL_THREAD)
+-		t_error("pthread_barrier_wait failed: got %d \"%s\", wanted either 0 or %d\n",
+-			r, strerror(r), PTHREAD_BARRIER_SERIAL_THREAD);
++	if(!(r && r != PTHREAD_BARRIER_SERIAL_THREAD))
```

Comment:
Maybe this if statement should keep the previous condition. I am not sure how it should be, but to me, it makes more sense the other way around.
```suggestion
+	if(r && r != PTHREAD_BARRIER_SERIAL_THREAD)
```

Reply:
Changed it.